src/box/lua/schema.lua

```diff
@@ -336,6 +336,14 @@ end
 
 box.is_in_txn = builtin.box_txn
 
+box.txn_id = function()
+    local id = tonumber(builtin.box_txn_id())
+    if id == -1 then
+        box.error(box.error.ACTIVE_TRANSACTION)
```

Comment:
It's not good error for this case. It's error for the case when you try to start transaction, but there is some other active transaction.